src/FixedDataTableCellGroup.js

```diff
@@ -74,24 +136,47 @@ class FixedDataTableCellGroupImpl extends React.Component {
     this._initialRender = false;
   }
 
+  /**
+   * Returns Object consisting of keys and widths of the columns in the current cell group.
+   * @returns {{keys: [], widths: []}}
+   */
+  getCellGroupWidth = () => {
+    const { columns } = this.props;
+    const cellGroupColumnWidths = {
+      keys: [],
+      widths: [],
+    };
+    if (this.props.isHeader) {
+      for (let i = 0, j = columns.length; i < j; i++) {
+        const key = columns[i].props.columnKey || 'cell_' + i;
+        cellGroupColumnWidths.keys.push(key);
+        cellGroupColumnWidths.widths.push(columns[i].props.width);
+      }
+    }
+    return cellGroupColumnWidths;
+  };
+
   render() /*object*/ {
     var props = this.props;
     var columns = props.columns;
     var cells = new Array(columns.length);
     var contentWidth = sumPropWidths(columns);
 
-    var isColumnReordering = props.isColumnReordering && columns.reduce(function(acc, column) {
-      return acc || props.columnReorderingData.columnKey === column.props.columnKey;
-    }, false);
-
     var currentPosition = 0;
     for (var i = 0, j = columns.length; i < j; i++) {
       var columnProps = columns[i].props;
       var cellTemplate = columns[i].template;
-      var recyclable = columnProps.allowCellsRecycling && !isColumnReordering;
-      if (!recyclable || (
-        currentPosition - props.left <= props.width &&
-        currentPosition - props.left + columnProps.width >= 0)) {
+
+      var recyclable = _.get(
+        this.state.recycling,
+        [columnProps.columnKey],
+        columnProps.allowCellsRecycling
+      );
```

Comment:
Recyclable columns have the performance benefit that cells within it get unmounted once they go out the visible viewport. So it serves as a performance improvement when the table has a large no:of columns, particularly during vertical scrolling.

But during reordering, if the cell being dragged is within a recyclable column, and the table gets scrolled horizontally such that the parent column goes outside the viewport, then all the cells within the column get unmounted, including the one being dragged.
This means, reordering won't work as expected.

The earlier code seems to have fixed it by treating any columns currently being reordered as non-recyclable, even if the column prop allowCellsRecycling is set.

Now, for backward compatibility we are storing that information in local state, but for newer users, they have to manually handle this case like (this)[].